datasets/scielo/scielo.py

```diff
@@ -94,24 +93,25 @@ def _split_generators(self, dl_manager):
             datasets.SplitGenerator(
                 name=datasets.Split.TRAIN,
                 gen_kwargs={
-                    "source_file": os.path.join(data_dir, f"{fname}.{lang_pair[0]}"),
-                    "target_file": os.path.join(data_dir, f"{fname}.{lang_pair[1]}"),
+                    "source_file": f"{fname}.{lang_pair[0]}",
+                    "target_file": f"{fname}.{lang_pair[1]}",
+                    "files": dl_manager.iter_archive(archive),
                 },
             ),
         ]
 
-    def _generate_examples(self, source_file, target_file, target_file_2=None):
-        with open(source_file, encoding="utf-8") as f:
-            source_sentences = f.read().split("\n")
-        with open(target_file, encoding="utf-8") as f:
-            target_sentences = f.read().split("\n")
+    def _generate_examples(self, source_file, target_file, files, target_file_2=None):
+        for path, f in files:
+            if path == source_file:
+                source_sentences = f.read().decode("utf-8").split("\n")
+            elif path == target_file:
+                target_sentences = f.read().decode("utf-8").split("\n")
+            elif self.config.name == "en-pt-es" and path == target_file_2:
+                target_sentences_2 = f.read().decode("utf-8").split("\n")
```

Comment:
This does download everything (up to 300MB) before starting to yield examples but it's better than nothing

Reply:
Yes, not ideal indeed. I'll try to create a PR that replaces the `file.read().decode("utf-8")` pattern with `enumerate(file)` when processing rows of data for all the datasets when I find more time.